src/Components/Web.JS/src/Services/NavigationEnhancement.ts

```diff
@@ -118,7 +118,12 @@ function onDocumentSubmit(event: SubmitEvent) {
 
     event.preventDefault();
 
-    const url = new URL(formElem.action);
+    // If the form has no action attribute, or an empty string, we do not want the default browser
+    // behavior of submitting to "current URL" because that would be wrong if there's still a pending
+    // enhanced nav in progress, e.g., after the user clicks "back" but before it loads. So in this
+    // case we manually supply the URL corresponding to the content on the screen, which will be the
+    // URL whose response rendered the form.
+    const url = new URL(formElem.getAttribute('action') || currentContentUrl);
```

Comment:
This is the entirety of the fix. The rest of the PR is removing redundant code. The existing E2E test still applies and shows this works.